### PR TITLE
Hapi Fixes to IServerInject

### PR DESCRIPTION
**Pull request by StanleyGoldman**

### CODE REVIEW

**Comment by jasonswearingen** [comment]
I don't agree with the af99a74 checkin.    **Reason**: switching from ```bluebird``` to ```es6-promise``` is going to break compatibility for everyone who currently uses ```hapi.d.ts```.   **Alternative Solution**: If you don't like the bluebird ref, better to add an explicit duck-typed promise interface.

other than that, I think everything is great.

**Comment by jasonswearingen** [comment]
oh, so I guess I should vote with thumbs for the bot:  :-1:

**Comment by jasonswearingen** [comment]
If it's too difficult to do the alternative solution, let me know and I can give it a go in the next day or so.

**Comment by StanleyGoldman** [comment]
Hey @jasonswearingen I added an internal IPromise<T> interface, but I didn't add any methods to it.

Should I make this object look like the promise of es6?

**Comment by jasonswearingen** [comment]
It would be nice to have proper IPromise like typings, but not so important.   

This will cause some typing breaks for people who upgrade, but it's not a super big deal, as they can cast the return to their "real" promise interface manually.

Sorry I don't really know how to contribute to your pending checkins, otherwise i'd work adding full duck typing myself.    I'll try filling it out later I suppose.

## So in summary ##   
Pass!  :+1:

**Comment by StanleyGoldman** [comment]
@jasonswearingen I don't mind adding it, I just wanted to be sure I was doing the right thing.
The non-DRY of what we need to do here is giving me pause.

So should I base the duck typed promise off es6 promises or bluebird? es6 right?

**Comment by jasonswearingen** [comment]
You are right that it's rather....   frustrating.... why we need to effectively copy/paste typings like this.    There are a ton of issues with d.ts versioning that will hopefully be resolved soon in typescript (here's an example thread: https://github.com/Microsoft/TypeScript/issues/5537

It seems the best way to be compatible right now is to minimize the number of d.ts references to the minimum.   

### as for the af9d89e checkin, it looks great! :+1:

**Comment by vvakame** [comment]
thanks mate!
